lib/opencensus/absinthe/middleware/dataloader.ex

```diff
@@ -0,0 +1,72 @@
+if Code.ensure_loaded?(Dataloader) do
+  defmodule Opencensus.Absinthe.Middleware.Dataloader do
+    @moduledoc """
+    This is a small extension on top of `Absinthe.Middleware.Dataloader` that
+    will create spans for each resolution.
+
+    ## Usage
+
+    In your Absinthe schema, simply override the `plugins/0` callback (if you're
+    not already) and prepend this plugin to the list:
+
+        def plugins do
+          [Opencensus.Absinthe.Middleware.Dataloader | Absinthe.Plugin.defaults()]
+        end
+    """
+
+    @behaviour Absinthe.Middleware
+    @behaviour Absinthe.Plugin
+
+    @span_key :dataloader_resolution_span_ctx
+    @counter_key :dataloader_resolution_counter
+
+    alias Opencensus.Absinthe.Acc
+    alias Absinthe.Middleware.Dataloader, as: DefaultDataloader
+
+    @doc """
+    The `Absinthe.Plugin` callback. Starts the OpenCensus span.
+    """
+    def before_resolution(exec) do
+      span_options = %{attributes: %{}}
+      acc = Acc.get(exec)
+
+      {counter, new_acc} =
+        Map.get_and_update(acc, @counter_key, fn cur ->
+          case cur do
+            nil -> {cur, 1}
+            x -> {x, x + 1}
+          end
+        end)
+
+      span_ctx = :oc_trace.start_span("resolution_#{counter || 0}", acc.span_ctx, span_options)
+      new_acc = Map.put(new_acc, @span_key, span_ctx)
+
+      exec
+      |> Acc.set(new_acc)
+      |> DefaultDataloader.before_resolution()
+    end
```

Comment:
I recently automated insertion of  `Opencensus.Absinthe.Middleware` on literally every field with a `resolver` in our code base, simulating what'll happen with Absinthe 1.5 by default. 

Turns out, it's a mess when tracing every field meets `Absinthe.Resolution.Helpers.dataloader/3`. For every item in the output, we get a span starting just before the call to `Dataloader.load` and ending just after the call to the last `on_load` in the chain. We need to know how long `Dataloader.run/1` took and what it was doing, but the trace can't tell us.

Strikes me if you finish your span right here, we'll know! 

Read also, #26, which is all about this.